xsnap/documentation/lockdown.md

```diff
@@ -0,0 +1,238 @@
+## lockdown()
+
+The `lockdown` function freezes all JS built-ins.
+
+- Once the `lockdown` function has been called, the `harden` function can be called.
+- Snapshots preserve the state of the machine, pre and post lockdown.  
+
+### Surgery
+
+Besides freezing all JS built-ins, the `lockdown` function also modifies some built-ins as specified by Secure ECMAScript.
+
+- The `constructor` property of the prototypes of function, generator and compartment instances are replaced by a function that always throws.
+
+In compartments created after the execution of the `lockdown` function:
+
+- The `Date` constructor is replaced by its secure version, which throws when called without arguments.
+- The `Date.now` and `Math.random` functions are replaced by their secure versions, which always throw.
```

Comment:
I recall that we agreed upon this divergence. Tracking https://github.com/endojs/endo/issues/934